app/controllers/application_controller.rb

```diff
@@ -466,6 +468,23 @@ def better_errors_allow_inline
     raise
   end
 
+  def current_ability
+    Ability.new(current_user).merge(granted_capability)
```

Comment:
This has lost the memoization of the default method, which has `@current_ability ||=` as a prefix - is that deliberate?

Reply:
Not deliberate by me. But adding in the memoization causes the tests to fail, and I would need to investigate why.

Reply:
The tests fail because in our controller tests, the controller state (e.g. instance variables) is not reset between each "request" - see https://github.com/rails/rails/issues/24566 . Calling `get` in a controller test isn't really making an http request, it just calls the methods on the controller object which exists for the length of the test.

So for a simple example, `site_controller_test#test_welcome` calls `get` twice. The memoized result from the first call means that the second logged-in call is also forbidden. This is, ultimately, because you're not really meant to make multiple requests in a single controller testcase - that should be in integration or system tests. But we do this a **lot** in our tests.